src/events/modify_queue_handler.rs

```diff
@@ -0,0 +1,75 @@
+use std::sync::Arc;
+
+use serenity::{
+    async_trait,
+    http::Http,
+    model::id::GuildId,
+    prelude::{Mutex, RwLock, TypeMap},
+};
+use songbird::{Call, Event, EventContext, EventHandler};
+
+use crate::{
+    client::GuildQueueInteractions,
+    commands::queue::{
+        build_nav_btns, calculate_num_pages, create_queue_embed, forget_queue_message,
+    },
+};
+
+pub struct ModifyQueueHandler {
+    pub http: Arc<Http>,
+    pub ctx_data: Arc<RwLock<TypeMap>>,
+    pub call: Arc<Mutex<Call>>,
+    pub guild_id: GuildId,
+}
+
+#[async_trait]
+impl EventHandler for ModifyQueueHandler {
+    async fn act(&self, ctx: &EventContext<'_>) -> Option<Event> {
+        if let EventContext::Track(_) = ctx {
```

Comment:
In reality, you only use this handler for the `Event::Track(TrackEvent::End)` so you know the `EventContext` will be of type Track. Since you don't use anything, I think you can remove the `if let`